contracts/examples/LiquidityExamples.sol

```diff
@@ -0,0 +1,238 @@
+
+    INonfungiblePositionManager public immutable nonfungiblePositionManager;
+
+    /// @notice Represents the deposit of an NFT
+    struct Deposit {
+        address owner;
+        uint128 liquidity;
+        address token0;
+        address token1;
+    }
+
+    /// @dev deposits[tokenId] => Deposit
+    mapping(uint256 => Deposit) public deposits;
+
+    constructor(INonfungiblePositionManager _nonfungiblePositionManager) {
+        nonfungiblePositionManager = _nonfungiblePositionManager;
+    }
+
+    // Implementing `onERC721Received` so this contract can receive custody of erc721 tokens
+    function onERC721Received(
+        address operator,
+        address,
+        uint256 tokenId,
+        bytes calldata
+    ) external override returns (bytes4) {
+        // get position information
+
+        _createDeposit(operator, tokenId);
+
+        return this.onERC721Received.selector;
+    }
+
+    function _createDeposit(address owner, uint256 tokenId) internal {
+        (, , address token0, address token1, , , , uint128 liquidity, , , , ) =
+            nonfungiblePositionManager.positions(tokenId);
+
+        // set the owner and data for position
+        // operator is msg.sender
+        deposits[tokenId] = Deposit({owner: owner, liquidity: liquidity, token0: token0, token1: token1});
+    }
+
+    /// @notice Calls the mint function defined in perphery, mints the same amount of each token.
+    /// For this example we are providing 1000 DAI and 1000 USDC in liquidity
+    /// @return tokenId The id of the newly minted ERC721
+    /// @return liquidity The amount of liquidity for the position
+    /// @return amount0 The amount of token0
+    /// @return amount1 The amount of token1
+    function mintNewPosition()
+        external
+        returns (
+            uint256 tokenId,
+            uint128 liquidity,
+            uint256 amount0,
+            uint256 amount1
+        )
+    {
+        // For this example, we will provide equal amounts of liquidity in both assets.
+        // Providing liquidity in both assets means liquidity will be earning fees and is considered in-range.
+        uint256 amount0ToMint = 1000;
+        uint256 amount1ToMint = 1000;
+
+        // transfer tokens to contract
+        TransferHelper.safeTransferFrom(DAI, msg.sender, address(this), amount0ToMint);
+        TransferHelper.safeTransferFrom(USDC, msg.sender, address(this), amount1ToMint);
+
+        // Approve the position manager
+        TransferHelper.safeApprove(DAI, address(nonfungiblePositionManager), amount0ToMint);
+        TransferHelper.safeApprove(USDC, address(nonfungiblePositionManager), amount1ToMint);
+
+        INonfungiblePositionManager.MintParams memory params =
+            INonfungiblePositionManager.MintParams({
+                token0: DAI,
+                token1: USDC,
+                fee: poolFee,
+                tickLower: TickMath.MIN_TICK,
+                tickUpper: TickMath.MAX_TICK,
```

Comment:
i believe these values won't work for all tick spacings? maybe just worth noting in a comment